test-data/unit/check-flags.test

```diff
@@ -1124,3 +1124,48 @@ GroupDataDict = TypedDict(  # type: ignore
 GroupsDict = Dict[str, GroupDataDict]  # type: ignore
 [builtins fixtures/dict.pyi]
 [out]
+
+[case testCheckDefaultAllowAnyGeneric]
+from typing import TypeVar, Callable
+
+T = TypeVar('T')
+C = Callable[[], T]
+
+def f(c: C):
+    pass
+[out]
+
+[case testCheckAllowAnyGenericAnyGeneric]
+# flags: --allow-any-generics
```

Comment:
I would instead test that setting both `--strict --allow-any-generics` (with strict being first) still allows them.

Reply:
Done.